File: microservice_case2/microservices/appointment-service/models/appointment.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Appointment struct {
	ID                 uint           `gorm:"primaryKey" json:"id"`
	PatientID          uint           `gorm:"index" json:"patient_id"`
	DoctorID           uint           `gorm:"index" json:"doctor_id"`
	AppointmentDate    time.Time      `gorm:"index" json:"appointment_date"`
	DurationMinutes    int            `json:"duration_minutes"`
	Status             string         `json:"status"` // scheduled, confirmed, completed, cancelled
	ReasonForVisit     string         `json:"reason_for_visit"`
	Notes              string         `json:"notes"`
	ConsultationType   string         `json:"consultation_type"` // in-person, video, phone
	MeetingLink        *string        `json:"meeting_link,omitempty"`
	CancelledAt        *time.Time     `json:"cancelled_at,omitempty"`
	CancellationReason *string        `json:"cancellation_reason,omitempty"`
	CreatedAt          time.Time      `json:"created_at"`
	UpdatedAt          time.Time      `json:"updated_at"`
	DeletedAt          gorm.DeletedAt `gorm:"index" json:"deleted_at,omitempty"`
}

type CreateAppointmentRequest struct {
	PatientID        uint    `json:"patient_id" binding:"required"`
	DoctorID         uint    `json:"doctor_id" binding:"required"`
	AppointmentDate  string  `json:"appointment_date" binding:"required"`
	ReasonForVisit   string  `json:"reason_for_visit" binding:"required"`
	ConsultationType string  `json:"consultation_type"`
	MeetingLink      *string `json:"meeting_link,omitempty"`
}

type UpdateAppointmentRequest struct {
	Status           *string `json:"status,omitempty"`
	ReasonForVisit   *string `json:"reason_for_visit,omitempty"`
	Notes            *string `json:"notes,omitempty"`
	ConsultationType *string `json:"consultation_type,omitempty"`
	MeetingLink      *string `json:"meeting_link,omitempty"`
}

type CancelAppointmentRequest struct {
	CancellationReason string `json:"cancellation_reason" binding:"required"`
}

File: monolit_case1/app/Models/Appointment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Appointment extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'patient_id',
        'doctor_id',
        'appointment_date',
        'duration_minutes',
        'status',
        'reason_for_visit',
        'notes',
        'consultation_type',
        'meeting_link',
        'cancelled_at',
        'cancellation_reason',
    ];

    protected $casts = [
        'appointment_date' => 'datetime',
        'cancelled_at' => 'datetime',
    ];

    /**
     * Get the patient
     */
    public function patient(): BelongsTo
    {
        return $this->belongsTo(User::class, 'patient_id');
    }

    /**
     * Get the doctor
     */
    public function doctor(): BelongsTo
    {
        return $this->belongsTo(User::class, 'doctor_id');
    }

    /**
     * Get prescriptions from this appointment
     */
    public function prescriptions(): HasMany
    {
        return $this->hasMany(Prescription::class);
    }

    /**
     * Scope for upcoming appointments
     */
    public function scopeUpcoming($query)
    {
        return $query->where('appointment_date', '>', now())
            ->whereIn('status', ['scheduled', 'confirmed']);
    }

    /**
     * Scope for completed appointments
     */
    public function scopeCompleted($query)
    {
        return $query->where('status', 'completed');
    }
}
